src/test/java/com/zebrunner/carina/demo/WebMyFitnessPal.java

```diff
@@ -0,0 +1,220 @@
+
+        MyFitnessPalLogInPage logInPage = new MyFitnessPalLogInPage(driver);
+        logInPage.open();
+
+        MyFitnessPalSignUpPage signUpPage = logInPage.goToSignUp();
+
+        //Check if Sign Up page was opened
+        Assert.assertTrue(signUpPage.isPageOpened());
+
+        signUpPage.clickContinue();
+
+        //Check when click continue button redirect to enter firstname modal
+        Assert.assertTrue(getDriver().getCurrentUrl().contains("input-name"));
+
+        Goals[] goalsSelected = {Goals.GAIN_MUSCLE,Goals.MANAGE_STRESS,Goals.GAIN_WEIGHT};
+
+        //Just one of this three
+        LooseWeight[] looseWeightsOptionsSelected = {LooseWeight.DID_NOT_ENJOY_FOOD};
+        MaintainWeight[] maintainWeightsOptionsSelected ={MaintainWeight.COOKING_WAS_HARD_OR_TIME_CONSUMING};
+        GainWeight[] gainWeightsOptionsSelected = {GainWeight.GAIN_MUSCLE_FOR_GENERAL_FITNESS};
+
+        GainMuscle[] gainMusclesOptionsSelected = {GainMuscle.TONE_UP};
+        ModifyDiet[] modifyDietsOptionsSelected = {ModifyDiet.OTHERS};
+        ManageStress[] manageStressesOptionsSelected = {ManageStress.LISTEN_MUSIC};
+        IncreaseSteps[] increaseStepsOptionsSelected = {IncreaseSteps.LESS_1000};
+
+        ActivityLevel activityLevelsOptionSelected = ActivityLevel.NOT_VERY_ACTIVE;
+
+        WeeklyGoal weeklyGoalLooseWeightSelected = WeeklyGoal.LOSE_50KG;
+        WeeklyGoal weeklyGoalGainWeightSelected = WeeklyGoal.GAIN_25KG;
+
+        Account myAccount = new Account();
+        myAccount.setFirstName(R.TESTDATA.get("account.first_name"));
+
+        String birthday = R.TESTDATA.get("account.birthday");
+        String myGoalWeight;
+        String goalWeightLoosing = R.TESTDATA.get("account.goal_weight_loosing");
+        String goalWeightGaining = R.TESTDATA.get("account.goal_weight_gaining");
+
+        myAccount.setSex(Sex.MALE);
+        myAccount.setBirthday(formatToBirthday(birthday));
+        myAccount.setCountry(Country.United_States);
+        myAccount.setZipCode(R.TESTDATA.get("account.zipcode"));
+        myAccount.setHeightFeet(R.TESTDATA.get("account.height_feet"));
+        myAccount.setHeightInches(R.TESTDATA.get("account.height_inches"));
+        myAccount.setCurrentWeightLbs(R.TESTDATA.get("account.current_weight"));
+        myAccount.setUserName(R.TESTDATA.get("account.user_name"));
+        myAccount.setEmail(R.TESTDATA.get("account.email"));
+        myAccount.setPassword(R.TESTDATA.get("account.password"));
+
+        signUpPage.enterName(myAccount.getFirstName());
+        signUpPage.clickNextFromName();
+
+        signUpPage.selectGoalOption(goalsSelected);
+
+        if(goalsSelected.length!=3){
+            signUpPage.clickNextFromGoals();
+        }
+
+        signUpPage.clickNextBigStep();
+
+        Arrays.stream(goalsSelected).forEach(goal->{
+            switch (goal){
+                case LOSE_WIGHT:
+                    signUpPage.selectLoseWeightOption(looseWeightsOptionsSelected);
```

Comment:
Tests are not dynamic methods, you know in advance what goals you choose, so you can call the necessary methods step by step, I think this will be more compact than a switch case. it is also possible to create a generic method instead (selectLoseWeightOption, selectMaintainWeightOption, selectGainWeightOptionOption...)